### PR TITLE
Add 'SimplifyBooleanReturns' rule

### PR DESCRIPTION
**Pull request by ThrawnCA**
----

Make sure these boxes are checked before submitting your PR -- thank you!

- [ ] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

**Comment by ThrawnCA** [comment]
Hmm. I'm pretty sure that the SpotBugsBot is picking up existing issues, but I'll have a shot at fixing them.

**Comment by ThrawnCA** [comment]
@KengoTODA How far do you want me to go in making SpotBugsBot happy? I'm happy to keep fixing its findings, but clearly that's putting a lot of scope creep into this pull request.

File: spotbugs/src/main/java/edu/umd/cs/findbugs/ba/AbstractDominatorsAnalysis.java [hunk 1]
```diff
@@ -61,11 +61,7 @@
      */
     public AbstractDominatorsAnalysis(CFG cfg, final boolean ignoreExceptionEdges) {
         this(cfg, edge -> {
```
[the remark this answers is not in the document]
Comment (ThrawnCA) [reply spotbugs/src/main/java/edu/umd/cs/findbugs/ba/AbstractDominatorsAnalysis.java]: @KengoTODA It appears that SonarQube chokes on Lambda syntax.

[the remark this answers is not in the document]
Comment (KengoTODA) [reply spotbugs/src/main/java/edu/umd/cs/findbugs/ba/AbstractDominatorsAnalysis.java]: I think it's reasonable, `edge -> !(ignoreExceptionEdges && edge.isExceptionEdge())` seems enough.

**Comment by KengoTODA** [comment]
> @KengoTODA How far do you want me to go in making @spotbugs-bot happy? 

If I was you, I will not mix up fixes that is not related with your new PMD rule. Even though spotbugs-bot reports some, Admin account like me can merge your changes.

**Comment by ThrawnCA** [comment]
OK, thanks. I'll wait and give humans a chance to review before I push any more.

**Comment by ThrawnCA** [comment]
@spotbugs/everyone Anyone have time to look over this?

**Pull request merged**